### DISCUSSION TITLE
filtering submitted language

### DISCUSSION THREAD

**Discussion by 0MAL** [mosparo/mosparo#215] [Q&A]
It looks like impossible to filter post that is NOT include some characters of language.
For example, when website is operated by Chinese language and a user send post that doesn't include single Chinese characters at all, then it is not normal and highly possible to be spam, vise versa.

Rule types of regex on mosparo doesn't allow not-operator, but it would be appreciated if not-operator regex can be used in the future.

This may be related to this topic:
[ Not in this country rule ] 
https://github.com/orgs/mosparo/discussions/208

**Comment by zepich** [comment]
Hi @0MAL 

Thank you very much for your suggestion.

Yes, you're right. It's related to #208.

But the good news is that a change will come in the next minor release (v1.2). Last weekend, when I was finishing the development of v1.2, I thought about #208 and whether there is an easy fix that we could add to it without changing a lot of stuff. While thinking about it, I discovered a pretty simple solution that will work for your case, too.

Right now, you can define a spam rating value for every rule item. This spam rating value ranges from `0.1` to `unlimited`. While thinking about that last weekend, I asked myself why we don't allow negative values here. What can you do when you want to 'reward' a word, an IP address, a country, or a Unicode Block? You don't have an option right now, but by allowing negative numbers, you can do that.

So the fix I added last weekend to v1.2 is that you can specify negative numbers, too. The new range of the spam rating field for every rule item is `-1000000` to `1000000`. As soon as v1.2 is released, you can specify a RegEx or even a Unicode Block for Chinese characters and give it a rating of `-10`, for example.

Since the spam score of a project must be 0.1 or higher, you would have to add an additional rule item, for example, an IP subnet rule item for `0.0.0.0/0` (which would include every IP address worldwide), and define to that, for example, the spam rating value of `1`. If someone submits a form with Chinese characters, the calculation is `1 + (-10) = -9`, so it's below the spam score value of 0.1 and, therefore, no spam. (see #212 for a better explanation)

The "bad" news is that v1.2's release will take some time. We expect the first beta to be available around mid-April and the official release at the end of May/Beginning of June.

Kind regards,

zepich
